d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -211,8 +219,102 @@ public void onSuccess(None result)
+    };
+
+    final Callback<None> warmupMarkDownCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e) {
+        _log.info("mark down warm up cluster " + _warmupClusterName + " has failed");
+        // what to do here ? - go ahead continue to announce main cluster
+        // doAnnounceCallback.onSuccess(None.none());
+      }
+
+      @Override
+      public void onSuccess(None result) {
+        _log.info("mark down warm up cluster " + _warmupClusterName +
+            " has completed, now mark up of main cluster " + _cluster);
+        doMarkUpCallback.onSuccess(None.none());
+      }
+    };
+
+    final Callback<None> doWarmupMarkDownCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e) {
+        _log.info("failure in scheduling warm up mark down");
+        // what to do here ? - go ahead continue to announce main cluster
+        // doAnnounceCallback.onSuccess(None.none());
+      }
+
+      @Override
+      public void onSuccess(None result) {
+        _log.info("warm up for cluster " + _warmupClusterName + " completed, now calling mark down on the warm up cluster");
+        _server.markDown(_warmupClusterName, _uri, warmupMarkDownCallback);
+      }
+    };
+
+    final Callback<None> doWarmupWaitCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e)
+      {
+        _log.info("failure in mark up of warm up cluster " + _warmupClusterName);
+        // what to do here
+        // 1. go ahead continue to announce main cluster
+        // doAnnounceCallback.onSuccess(None.none());
+        // OR 2. mark announcement fail, move on to next pending mark up / mark down
+        // announceCallback.onError(e);
+      }
+
+      @Override
+      public void onSuccess(None result)
+      {
+        _log.info("mark up for warm up cluster " + _warmupClusterName + " succeeded");
+        ScheduledExecutorService warmup = Executors.newSingleThreadScheduledExecutor();
+        _log.info("scheduled executor service to run after warmup");
+        warmup.schedule(() -> doWarmupMarkDownCallback.onSuccess(None.none()), _warmupDuration, TimeUnit.MILLISECONDS);
+      }
+    };
+
+    final Callback<None> doWarmupMarkUpCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e)
+      {
+        // this would ideally never happen
+        callback.onError(e);
+      }
+
+      @Override
+      public void onSuccess(None result)
+      {
+        _log.info("Calling ZooKeeperServer's mark up for warm up cluster " + _warmupClusterName);
+        _server.markUp(_warmupClusterName, _uri, _partitionDataMap, _uriSpecificProperties, doWarmupWaitCallback);
+      }
+    };
+
     _log.info("overrideMarkUp is called for uri = " + _uri);
+    if (_isEnabledD2WarmupCluster) {
+      _log.info("Calling mark up for warm up cluster " + _warmupClusterName);
+      doWarmupMarkUpCallback.onSuccess(None.none());
+    } else {
```

Comment:
Looks like the style in this file is to have the curly braces surrounding "else" on separate lines.  It would be good to match the existing style & move your curly braces down onto the next lines

Reply:
Thanks! Fixed.